SingleMeshBlend/SingleMeshBlend/SingleBlendMeshDeformer.cpp

```diff
@@ -204,37 +204,44 @@ MThreadRetVal SingleBlendMeshDeformer::threadEvaluate(void * pParam)
 	unsigned int start{ threadData->start };
 	unsigned int end{ threadData->end };
 
-	MPointArray& vertexPositions{ data->vertexPositions };
-	MPointArray& blendVertexPositions{ data->blendVertexPositions };
+	//Operator [] on maya containers as  a non negligible overhead. By using pointer we can bypass it
+	MPoint* currentVertexPosition{ &data->vertexPositions[0] };
+	MVectorArray deltas{ data->deltas };
 
 	float envelopeValue{ data->envelopeValue };
 	double blendWeightValue{ data->blendWeightValue };
 
-	//MPointArray operator[] has a non-negligible overhead. 
-	//Accessing the memory directly with pointers bypass that overhead.
-	unsigned int vertexCount{ vertexPositions.length() };
-	MPoint* currentVertexPosition{ &vertexPositions[0] };
-	MPoint* blendVertexPosition{ &blendVertexPositions[0] };
-
-	for (unsigned int vertexIndex{ start }; vertexIndex < end; vertexIndex++) {
-		MVector delta{ (*(blendVertexPosition + vertexIndex) - *(currentVertexPosition + vertexIndex)) * blendWeightValue * envelopeValue };
-		MPoint newPosition{ delta + *(currentVertexPosition + vertexIndex) };
-
-		*(currentVertexPosition + vertexIndex) = newPosition;
+	unsigned int vertexCount{ data->vertexPositions.length() };
+	for (unsigned int vertexIndex{ start }; vertexIndex < end; ++vertexIndex) {
+		currentVertexPosition[vertexIndex] = (deltas[vertexIndex] * blendWeightValue * envelopeValue) + currentVertexPosition[vertexIndex];
```

Comment:
remember to change this to 3 lines each computing a coordinate, so you can use a double* instead of a MPoint* . You are still using the MPoint and vector operators.

Reply:
Yeah, this was actually in the last commit that I didn't push before going out.